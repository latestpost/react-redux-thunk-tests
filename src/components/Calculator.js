import React from "react";
import Button from './Button';
import { addInput, multiplyInput } from '../actions/calculatorActions';
import { connect } from 'react-redux';

class CalculatorContainer extends React.Component {
  render() {
    return (
        <div>
            <Button onClick={this.props.handleAdd} buttonValue='Add 300'></Button><br />
            <Button onClick={this.props.handleMultiply} buttonValue='Multiply by 10'></Button><br />
            <div>{this.props.state.output}</div>
        </div>
    )
  }
} 
const mapStateToProps =  (state) => {
  return {state};
}

const mapDispatchToProps = (dispatch) => {
  return {
    handleAdd(e) {
      dispatch(addInput(300));
    },
    handleMultiply(e) {
      dispatch(multiplyInput(10));
    }
  }
};

const Calculator = connect(mapStateToProps, mapDispatchToProps)(CalculatorContainer);
export default Calculator;