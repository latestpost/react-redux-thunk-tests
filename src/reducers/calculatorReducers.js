const initialState = {output:0}

const calculatorReducers = (state=initialState,action) => {
    switch (action.type){      
        case 'ADD_INPUT': 
            return {...state, output:action.input + state.output}
        case 'MULTIPLY_INPUT':
            return {...state,output:action.input * state.output}
        default: return {...state}
    }
}

export default calculatorReducers