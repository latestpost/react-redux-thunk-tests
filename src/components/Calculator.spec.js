import React, {Component} from 'react';
import { Provider, connect } from 'react-redux';
import reducers from '../reducers/calculatorReducers';
import { createStore } from 'redux';

import toJson from 'enzyme-to-json';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

const store = createStore(reducers);

import Calculator from './Calculator';

it('renders correctly', () => {
  const wrapper = render(
    <Provider store={store}>
        <Calculator />
    </Provider>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});