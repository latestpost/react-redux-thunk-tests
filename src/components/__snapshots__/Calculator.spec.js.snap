// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button>
    Add 300
  </button>
  <br>
    
  </br>
  <button>
    Multiply by 10
  </button>
  <br>
    
  </br>
  <div>
    0
  </div>
</div>
`;
